client/src/lib/dependency-injection/inject.js

```diff
@@ -1,24 +1,29 @@
 import React from 'react';
 import injectorContext from './injectorContext';
 
-const inject = (dependencies, mapDependenciesToProps, contextArg) => (Component) => {
+/**
+ * Injects dependences from the Injector as named props into a component
+ * @param {Array} dependencies A list of named dependencies
+ * @param {callback} mapDependenciesToProps Maps the fetched dependencies to propnames, e.g.
+ *   (TextField, FormAction) => { myText: TextField, myAction: FormAction }
+ * @param {function} getContext Gets the name of the context to pass to Injector.
+ *   Accepts the component's props as a param. (props) => `SomeContext.${props.identifier}`;
+ */
+const inject = (dependencies, mapDependenciesToProps, getContext) => (Component) => {
   // eslint-disable-next-line react/prefer-stateless-function
   class Injected extends React.Component {
     render() {
       let props = {};
-      const context = typeof contextArg === 'function' ?
-        contextArg(this.props) :
-        contextArg;
-      let deps = dependencies;
+      const context = getContext && typeof contextArg === 'function' ?
```

Comment:
could have a `getContext && typeof getContext !== 'function'` check above this to throw an exception, but not necessary.
this could then be simplified to `const context = (getContext) ? getContext(this.props) : null;`